exploration/2025/gf_reanalysis_fs.md

```diff
@@ -0,0 +1,433 @@
+```
+
+```python
+df_ra = glofas.load_glofas_reanalysis(station_name=station_name)
+```
+
+```python
+df_ra
+```
+
+```python
+adm2 = codab.load_codab_from_blob(admin_level=2, aoi_only=True)
+```
+
+```python
+pcodes = BENUE_ADM2_PCODES
+```
+
+```python
+query = f"""
+SELECT *
+FROM public.floodscan
+WHERE pcode IN {tuple(pcodes)}
+"""
+```
+
+```python
+df_fs = pd.read_sql(
+    query, stratus.get_engine(stage="prod"), parse_dates=["valid_date"]
+)
+```
+
+```python
+df_sfed = df_fs[df_fs["band"] == "SFED"]
+```
+
+```python
+blob_name = f"{blob.PROJECT_PREFIX}/processed/floodscan/fs_benue_pixels_1998_2024.parquet"
+df_fs_pixels = stratus.load_parquet_from_blob(blob_name)
+```
+
+```python
+df_fs_pixels["pixel_id"] = (
+    df_fs_pixels["x"].astype(str) + "_" + df_fs_pixels["y"].astype(str)
+)
+```
+
+```python
+df_fs_pixels
+```
+
+## Compare pixels
+
+See which pixels have the best correlation with GloFAS reanalysis
+
+```python
+df_compare_pixels = df_ra.rename(columns={"time": "date"}).merge(df_fs_pixels)
+```
+
+```python
+dicts = []
+for (x, y), group in df_compare_pixels.groupby(["x", "y"]):
+    dicts.append(
+        {
+            "x": x,
+            "y": y,
+            "corr": group.corr(numeric_only=True).loc["dis24", "SFED"],
+        }
+    )
+```
+
+```python
+df_compare_pixels.groupby("date")[["SFED", "dis24"]].mean().corr()
+```
+
+```python
+df_corr_pixels = pd.DataFrame(dicts).sort_values("corr", ascending=False)
+```
+
+```python
+df_corr_pixels.set_index(["y", "x"]).to_xarray()["corr"].plot()
+```
+
+```python
+df_corr_pixels.sort_values("corr", ascending=False)
+```
+
+```python
+top_x, top_y = df_corr_pixels.sort_values("corr", ascending=False).iloc[1][
+    ["x", "y"]
+]
+```
+
+## Combine reanalysis with Floodscan
+
+There are a few different ways we can select with Floodscan pixels to use in the comparison.
+
+Run only the relevant cells to merge that Floodscan aggregation for comparison.
+
+### 1. Use admin2 Floodscan raster stats
+
+```python
+df_compare = (
+    df_ra.rename(columns={"time": "valid_date"})
+    .merge(df_sfed.rename(columns={"pcode": "ADM2_PCODE"}))
+    .merge(adm2[["ADM2_PCODE", "ADM2_EN"]])
+)
+```
+
+```python
+for name, group in df_compare.groupby("ADM2_EN"):
+    print(name)
+    display(group.corr(numeric_only=True).loc["dis24", "mean"])
+```
+
+```python
+df_compare_adm2 = df_compare[df_compare["ADM2_PCODE"] == NUMAN2].copy()
+```
+
+### 2. Use mean of pixels
+
+```python
+df_compare_adm2 = (
+    df_compare_pixels.groupby("date")[["dis24", "SFED"]]
+    .mean()
+    .reset_index()
+    .copy()
+    .rename(columns={"date": "valid_date", "SFED": "mean"})
+)
+```
+
+### 3. Use the best pixel
+
+```python
+df_compare_adm2 = (
+    df_compare_pixels[
+        (df_compare_pixels["x"] == top_x) & (df_compare_pixels["y"] == top_y)
+    ]
+    .copy()
+    .rename(columns={"date": "valid_date", "SFED": "mean"})
+)
+```
+
+## Further processing
+
+```python
+df_compare_adm2["mean"] = (
```

Comment:
Hmmm a good point here to take the rolling average of Floodscan data. I should probably do this in my analysis too